packages/extensions-framework/src/ExtensionProvider.ts

```diff
@@ -0,0 +1,88 @@
+import { injectable, interfaces } from "inversify";
+import { BehaviorSubject } from "rxjs";
+
+import Container from "./Container";
+
+export interface IExtensionProvider<T> {
+  subscribe(callback: () => void): any;
+  getAllExtensions(): T[];
+  getTaggedExtensions(tagName: string, tagValue: any): T[];
+}
+
+@injectable()
+export class ExtensionProvider<T> implements IExtensionProvider<T> {
+  private serviceIdentifier: interfaces.ServiceIdentifier<T>;
+  private container: Container;
+  private extensions$: BehaviorSubject<any>;
+
+  constructor(
+    serviceIdentifier: interfaces.ServiceIdentifier<T>,
+    container: Container
+  ) {
+    this.serviceIdentifier = serviceIdentifier;
+    this.container = container;
+    this.extensions$ = new BehaviorSubject<any>(null);
+
+    this.container.addEventListener<
+      T
+    >(
+      Container.BOUND,
+      (
+        identifier: interfaces.ServiceIdentifier<T>
+      ) => {
+        if (identifier === this.serviceIdentifier) {
+          this.extensions$.next(identifier);
+        }
+      }
+    );
+  }
+
+  public subscribe(callback: () => void) {
```

Comment:
The callback surely gets something, it seems like it gets at least `interfaces.ServiceIdentifier<T>` as param